frontend/src/errors.ts

```diff
@@ -30,14 +29,14 @@ export class NetworkError extends Error {
     const serializedSearchInputs = searchInputs
       ? convertSearchInputSetsToArrays(searchInputs)
       : {};
-
-    const serializedData = JSON.stringify({
+    const message = error instanceof Error ? error.message : "Unknown Error";
+    const cause = {
       type: "NetworkError",
       searchInputs: serializedSearchInputs,
-      message: error instanceof Error ? error.message : "Unknown Error",
+      message,
       status: 500,
-    });
-    super(serializedData);
+    };
+    super(message, { cause });
```

Comment:
Adds the details to the [cause](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause) property which serializes it.